### PR TITLE
OOP - Milestone: game logic

### PR DESCRIPTION
**Pull request by jebitok-dev**
In this milestone I:-
- set-up the project repository and files
- implemented the game logic using OOP programming methods i.e combining blocks, classes, DRY principle, inheritance, and scope, among others
- added Linters and gems files

### CODE REVIEW

**Comment by jebitok-dev** [comment]
I will resolve the conflicts on the branch before merging to develop. (I will have to merge to the branches on Command Line for it to be resolved.)

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUIRED
Hello team,
Nice work implementing the game logic. However some changes are required.

[OPTIONAL] Break your logic into classes to group information efficiently. e.g: create a class Player having player info, like names, etc. This helps in a better implementation of the game

[REQUIRED] In main.rb, remove return puts on line 27, 28 and 31

Implement the changes and ask for another review

**Review by MarilenaRoque — CHANGES_REQUESTED** [review]
## Hello! Congrats on your implementation of the game logic. :rocket:

However, let's refactor the part of the code to assure the use of the best practices.

### ♻️ Required Changes
- [ ] [REQUIRED]You can use **break** to break the loop and avoid the unnecessary use of explicit return.
- [ ] [Optional] When you have multiples excluding conditions is more appropriated to use **elsif** instead of different **if** blocks.

- Don't forget to solve the conflicts to be able to merge your branch.

Any questions about the review? Send me a message on Slack 👍🏻 
## Happy Coding! 💻 

**_Marilena Roque
Full-Stack Developer
Code Reviewer at Microverse Inc.
[Portfolio](https://mariroqueproject.com/) | [Linkedin](https://www.linkedin.com/in/roquemarilena/) 
marilenas@id.uf.br_**

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  end
+
+  def description
+    puts "#{@name} has won"
+  end
+
+  def details
+    puts 'Game is a draw'
+  end
+end
```
Comment (cristianCeamatu) [inline lib/player.rb]: - [ ] [Please fix] No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). In order to fix this you can define the two methods in the `main.rb` file similar to how you defined the `player_move` method. Of course, `description` should accept a name parameter.

**Review by cristianCeamatu — APPROVED** [review]
## 🥇  APPROVED
Congratulations guys! The game works perfectly and passes all the project requirements. I am impressed with how simple your project was to read and understand all the core functionality. I only found an issue and I did not want to request changes to your project for this because you already waited for a couple of reviews. Please check my inline comment with the [Please fix] before merging the request.  Also check the suggestions, even if you don't implement them at least you know that you can use such functionality with a future project.

### Suggestions
1. [Optional] Add `sleep` and colors to your game, this will totally transform your game and make it feel like it is not a simple CLI game. Links: [colors (the second answer is without using any gem)
](https://stackoverflow.com/questions/1489183/colorized-ruby-output-to-the-terminal), [sleep](https://stackoverflow.com/questions/1329967/tell-ruby-program-to-wait-some-amount-of-time) and a working example: [example](https://repl.it/@cristianCeamatu/microverse-ruby-tic-tac-toe)

### Please also check my inline comments below.

Good luck with future projects and congratulations on this project!  
# 👍 🥇 👨‍💻 

Best regards,
Cristian Ceamatu
Slack: (Search `Cristian Viorel Ceamatu`)

**Pull request merged**